include/exadg/convection_diffusion/time_integration/time_int_bdf.cpp

```diff
@@ -427,11 +428,11 @@ TimeIntBDF<dim, Number>::print_solver_info() const
 
 template<int dim, typename Number>
 void
-TimeIntBDF<dim, Number>::read_restart_vectors(boost::archive::binary_iarchive & ia)
+TimeIntBDF<dim, Number>::read_restart_vectors(BoostInputArchiveType & ia)
 {
   for(unsigned int i = 0; i < this->order; i++)
   {
-    ia >> solution[i];
+    read_write_distributed_vector(solution[i], ia);
```

Comment:
I would personally find it more intuitive to have two different functions `read_distributed_vector` (for this case) and `write_distributed_vector` for the `write_restart_vector` function below, given that we already have two different functions in this class and all similar classes below. There is the issue of code duplication, but to me the fact that the boost implementation can deduce from the templated type whether it should read or write is in my opinion a bigger obstacle for many users than duplicating 10 lines of code (one could clean up the functions a bit if we just read or write). The other question is whether we should implement a restart functionality in the deal.II vector classes, which would simplify this code here.

Reply:
thats one of the reasons why I chose this approach initially, I totally agree its easier to read `<<` and `>>` than to look up the docu of `boost` and find that it is handled automatically.
I have no strong preference as the explanations in the related dealii steps also use the `&` and explain it and we use it elsewhere. After Niklas' https://github.com/exadg/exadg/pull/722#issuecomment-2645793739  I changed it from what you are suggesting now.

Regarding putting vector serialization into dealii:
As of now, one usually goes the way through `SolutionTransfer` calling `prepare_for_serialization` and serialized the `triangulation` afterwards. Since I am working on this part at the moment, I would say we go with the current version and then look at the generalized version I come up with.

Reply:
OK, let's keep it for now and see where we'll be going.